### PR TITLE
Add feature to build multi-arch image

### PR DESCRIPTION
**Pull request by Divyansh42**
Since buildah now supports multi arch image built
support (ref: https://github.com/containers/buildah/issues/1590)
This feature will allow building images based on multiple
architectures.

Fixes: #24 

Signed-off-by: divyansh42 <diagrawa@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -116,6 +116,15 @@ envs: |
   GOPATH=/root/buildah/go</pre>
     </td>
   </tr>
+
+  <tr>
+    <td>archs</td>
+    <td>No</td>
+    <td>ARCH of the image to build, for multiple ARCHs seperate by a comma. For example, <code>"arm64,amd64"</code>.<br>
+    Defaults to <code>amd64</code>.<br>
+    If running on self hosted runners then suggest running <a href="https://github.com/marketplace/actions/docker-setup-qemu">qemu-user-static</a> action to install <code>qemu-user-static</code>.
```
Comment (tetchel) [inline README.md:125]: > The package `qemu-user-static` must be installed in order to build for another architecture. GitHub Linux runners already include this package, but if you are using a self-hosted runner, you can use the `docker-setup-qemu` action to make sure it is installed.

File: action.yml [hunk 2]
```diff
@@ -44,6 +44,10 @@ inputs:
     description: 'Set to true to build using the OCI image format instead of the Docker image format'
     default: 'false'
     required: false
+  archs:
+    description: 'ARCH of the image to build, for multiple ARCHs seperate by a comma. For example, "arm64,amd64"'
```
[the remark this answers is not in the document]
Comment (tetchel) [reply action.yml:48]: update the readme the same way

**Review by tetchel — CHANGES_REQUESTED** [review]

File: action.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Divyansh42) [reply action.yml:48]: spaces are not allowed after comma, but if user adds extra space I have handled that case in code.

File: src/index.ts [hunk 3]
```diff
@@ -20,12 +20,15 @@ export async function run(): Promise<void> {
     const tagsList: string[] = tags.split(" ");
     const newImage = `${image}:${tagsList[0]}`;
     const useOCI = core.getInput(Inputs.OCI) === "true";
+    let archs = core.getInput(Inputs.ARCHS);
+    // remove white spaces (if any) in archs input
+    archs = archs.replace(/\s+/g, "");
```
Comment (tetchel) [inline src/index.ts:25]: this looks to me like spaces are allowed after the comma because they will get removed here anyway

Comment (Divyansh42) [reply src/index.ts:25]: Yes, spaces are not allowed.

Comment (tetchel) [reply src/index.ts:25]: To clarify, above I was asking if the user could input spaces. I see that spaces are not allowed in the argument actually passed to buildah.

Comment (Divyansh42) [reply src/index.ts:25]: Yes, spaces are not allowed in the argument actually passed to buildah, to make it user friendly if by mistake user added any extra space then that will get removed.

**Comment by tetchel** [comment]
pinged you re: reasoning about multiple architectures

File: README.md [hunk 4]
```diff
@@ -115,7 +117,7 @@ jobs:
     - run: mvn package
 
     - name: Build Image
-      uses: redhat-actions/buildah-build@v2
+      uses: redhat-actions/buildah-build@v3
```
Comment (tetchel) [inline README.md:120]: should still be v2?

File: action.yml [hunk 5]
```diff
@@ -44,6 +44,12 @@ inputs:
     description: 'Set to true to build using the OCI image format instead of the Docker image format'
     default: 'false'
     required: false
+  archs:
+    description: |
```
Comment (tetchel) [inline action.yml:48]: since this is already a multiline block you don't have to quote it

File: README.md [hunk 6]
```diff
@@ -32,6 +33,7 @@ After building your image, use [push-to-registry](https://github.com/redhat-acti
 
 | Input Name | Description | Default |
 | ---------- | ----------- | ------- |
+| archs | Architecture(s) to build the image(s) for. For multiple architectures, separate by a comma. If running on self hosted runners then suggest running [qemu-user-static](https://github.com/marketplace/actions/docker-setup-qemu) action to install `qemu-user-static`. | `amd64`
```
Comment (tetchel) [inline README.md:36]: If the workflow is running on a self-hosted runner, [docker-setup-qemu] can be used to install the `qemu-user-static` dependency.

**Review by tetchel — CHANGES_REQUESTED** [review]

**Pull request merged**